src/main/java/com/sarality/db/SQLiteDatabaseProvider.java

```diff
@@ -63,5 +67,98 @@ public void onCreate(SQLiteDatabase database) {
   @Override
   public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
     logger.info("Upgrading tables for database {} to version {}.", dbName, dbVersion);
+
+    for (TableDefinition definition : definitionList) {
+      // get current schema
+      List<Column> versionColumns = new ArrayList<>();
+      boolean isSchemaInitialized = false;
+
+      for (Integer version = oldVersion + 1; version <= newVersion; version++) {
+        Map<UpdateOperationType, List<UpdateOperation>> operations = definition.getUpdatesForVersion(version);
+        if (operations == null) {
+          logger.info("No version upgrade operations for table {} in version {}.", definition.getTableName(), version);
+          continue;
+        }
+
+        logger.info("Running upgrade operations for table {} in version {}.", definition.getTableName(), version);
+
+        if (!isSchemaInitialized) {
+          versionColumns = getSchemaForVersion(oldVersion, definition);
+          isSchemaInitialized = true;
+        }
+
+        // is this a new table in this version?
+        List<UpdateOperation> updatesList = operations.get(UpdateOperationType.CREATE_TABLE);
```

Comment:
Might make sense to create a SchemaUpdateClassifier to manage the Schema Updates and then use that in both the VersionSchemaGenerator as well as here